src/skadi/sample.cc

```diff
@@ -195,15 +163,21 @@ const int16_t* sample::source(uint16_t index) const {
     return unzipped_cache.second.data();
   }
 
+  // for setting where to read compressed data from
+  auto src_func = [&mapped](z_stream& s) -> void {
+    s.next_in = static_cast<Byte*>(static_cast<void*>(mapped.second.get()));
+    s.avail_in = static_cast<unsigned int>(mapped.second.size());
+  };
+
+  // for setting where to write the uncompressed data to
+  auto dst_func = [this](z_stream& s) -> int {
+    s.next_out = static_cast<Byte*>(static_cast<void*>(unzipped_cache.second.data()));
+    s.avail_out = HGT_BYTES;
+    return Z_FINISH; // we know the output will hold all the input
```

Comment:
for this one we know exactly how large stuff should be so we can decompress it in one shot. this is why the old implement ion of gunzip was so succinct (no loop needed)